client/my-sites/site/index.jsx

```diff
@@ -227,6 +227,9 @@ export default React.createClass( {
 			'has-edit-capabilities': userCan( 'manage_options', site )
 		} );
 
+		// Fallback on site domain if site title is blank
+		const siteTitle = site.title && site.title.trim().length > 0 ? site.title : site.domain;
```

Comment:
this should already be happening when the site comes in through either the redux reducer or the flux action. It suggests that isn't happening correctly.